### PR TITLE
[FEATURE] MS-38: Candidates SignUp

### PR DESCRIPTION
**Pull request by JamirMansilla**
as the title says
**Commits**

- MS-38: create the layout to candidate´s signup page
- MS-38: add signups steps
- MS-38: add function to create candidate
- MS-38: refactor all the layout from signup
- 🚀

### CODE REVIEW

File: src/Components/Candidate/Profile/index.jsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            </form>
+          )}
+        />
+      </section>
+      {modal && <Modal />}
+    </>
+  );
+}
+
+export default CandidatesForm;
```
Comment (EugenioPonzone) [inline src/Components/Candidate/Profile/index.jsx]: Why have you brought this file?

File: src/Components/Candidate/Signup/signup2.jsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+              <Field name="mainSkills" label="Main Skilss" element="input" component={Fieldset} />
+              <Field
+                name="profileTypes"
+                label="Profile Types"
+                element="input"
+                component={Fieldset}
+              />
+              <Field name="education" label="Education" element="input" component={Fieldset} />
+              <Field name="experiences" label="Experiences" element="input" component={Fieldset} />
+              <Field name="courses" label="Courses" element="input" component={Fieldset} />
```
Comment (EugenioPonzone) [inline src/Components/Candidate/Signup/signup2.jsx:107]: These fields arent expected to be in this page.

File: src/Components/Candidate/Signup/signup2.jsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                element="input"
+                type="number"
+                component={Fieldset}
+              />
+              <Field
+                name="isOpenToWork"
+                label="Open to Work?"
+                element="input"
+                type="checkbox"
+                component={Fieldset}
```
Comment (EugenioPonzone) [inline src/Components/Candidate/Signup/signup2.jsx:126]: This one neither.

File: src/Components/Candidate/Signup/signup2.jsx (hunk 2 above)
Comment (EugenioPonzone) [inline src/Components/Candidate/Signup/signup2.jsx:107]: These fields arent expected to be in this page.

File: src/Routes/candidate.jsx [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
 
 const CandidatesRoutes = () => {
   const { url } = useRouteMatch();
   return (
     <Layout routes={candidatesRoutes} resource={'candidate'}>
       <Switch>
         <Route path={`${url}/home`} exact component={Home} />
-        <Route path={`${url}/sign-up`} exact component={Home} />
+        <Route path={`${url}/sign-up`} exact component={SignUp} />
         <Route path={`${url}/sign-up/step2`} exact component={Home} />
```
Comment (EugenioPonzone) [inline src/Routes/candidate.jsx:18]: If we dont use this route, I think this line could be removed

File: src/Components/Candidate/Signup/signup1.jsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  const modal = useSelector((store) => store.modal.show);
+
+  const submitForm = (formValues) => {
+    return dispatch(createCandidates(formValues));
+  };
+
+  const validate = (formValues) => {
+    const errors = {};
+    errors.email = validateEmail(formValues.email);
+    errors.password = validateText(formValues.password, 'Password', 8, 16);
```
Comment (EugenioPonzone) [inline src/Components/Candidate/Signup/signup1.jsx:22]: We have to validate that the 2 passwords are equals.

File: src/Components/Candidate/Signup/signup1.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+              <Field name="email" label="Email" element="input" type="email" component={Fieldset} />
+              <Field
+                name="password"
+                label="Password"
+                element="input"
+                type="password"
+                component={Fieldset}
+              />
+              <Field
+                name="password"
```
Comment (EugenioPonzone) [inline src/Components/Candidate/Signup/signup1.jsx:47]: This field have to have a different name than the field before.

File: src/Components/Candidate/Signup/index.jsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+          <h1 className={styles.mainTitle}>{FormTitle[page]}</h1>
+          {PageDisplay()}
+          <button
+            className={styles.buttonGreen}
+            disabled={page === 0}
+            onClick={() => {
+              setPage((currPage) => currPage - 1);
+            }}
+          >
+            Prev
```
Comment (EugenioPonzone) [inline src/Components/Candidate/Signup/index.jsx:30]: Try to follow the screenflow. In the sing up 1 you only need a button continue (type submit) and  in the sign up 2 you only need the button submit. 
The button continue in the first one must validate the form and if everything is ok, follow to the second one.

File: src/Routes/candidate.jsx (hunk 4 above)
Comment (JamirMansilla) [reply src/Routes/candidate.jsx:18]: done.

File: src/Components/Candidate/Signup/signup2.jsx (hunk 3 above)
Comment (JamirMansilla) [reply src/Components/Candidate/Signup/signup2.jsx:126]: done

File: src/Components/Candidate/Signup/signup2.jsx (hunk 2 above)
Comment (JamirMansilla) [reply src/Components/Candidate/Signup/signup2.jsx:107]: ok

File: src/Components/Candidate/Signup/signup1.jsx (hunk 6 above)
Comment (JamirMansilla) [reply src/Components/Candidate/Signup/signup1.jsx:47]: done

File: src/Components/Candidate/Signup/signup1.jsx (hunk 5 above)
Comment (JamirMansilla) [reply src/Components/Candidate/Signup/signup1.jsx:22]: done

File: src/Components/Candidate/Signup/index.jsx (hunk 7 above)
Comment (JamirMansilla) [reply src/Components/Candidate/Signup/index.jsx]: done

**Pull request closed without merging**